### ISSUE TITLE
doing a convolve2d in valid mode but return same size. 

### ISSUE THREAD

**Issue by nickylimjj** [scipy/scipy#9923]
I understand that there are 3 modes for convolving and correlating. For `valid` mode, it outputs a smaller image as it only convolves the valid elements. I believe it should be a core functionality to convolves in `valid` mode but output the same size image padded with `fill_value`

**Comment by rgommers** [comment]
> I understand that there are 3 modes for convolving and correlating.

For `signal.convolve/correlate`, indeed. `ndimage.convolve/correlate` always returns the same size, which is usually what you want for image processing type applications. what is your use case? I can imagine why your proposal would be useful, but only `fill_value=np.nan` would be a sensible default.

**Comment by nickylimjj** [comment]
My use case is for image convolution. I am a student in a computer vision class and we are using the `scipy.signal` library for `convolve2d` and wanted to get the same output size for `valid` convolution and pad the rest with `0`s.  Our hacky solution is to return a smaller image and embed it in a new image of original size. Perhaps it is better for the `ndimage` library, but it seems like that doesn't have a simple function call for that behavior.

**Comment by rgommers** [comment]
> Perhaps it is better for the ndimage library, but it seems like that doesn't have a simple function call for that behavior either.

Indeed. No one has asked for it before. The reason I think is simply that filling with zeros is arbitrary, and should be strictly worse than using `ndimage.convolve` with `mode='constant', cval=0`. That way you get results that are between zero and the "right" answer.

**Comment by rfezzani** [comment]
What about returning the values of the input array unchanged where the convolution is not 'valid'?
